technical-reports/format/composite-types.md

```diff
@@ -0,0 +1,220 @@
+In a similar vein, a GUI tool can "pluck" out the individual values of a composite token and use them as it would normally.
+
+E.g. a design tool like Figma might not have the concept of a color pair, so it can't do anything special with tokens of that type. However, that doesn't prevent it from displaying the foreground and background colors of those tokens alongside any plain color tokens in a color picker.
+
+### Appending custom type definitions
+
+While a tool that imports a token file containing user-defined type definitions that it doesn't understand might not be able to do anything special with them, there's nothing stopping tools that export or modify token files from adding their own type definitions to the file alongside tokens of that type.
+
+For example, imagine Sketch wanted to export all layer styles from a Sketch document to a token file. The Sketch app could have a built-in type definition that corresponds to layer styles. Perhaps something like this:
+
+<aside class="example" title="Custom type definitions for Sketch">
+
+```json
+{
+  "sketch layer style": {
+    "type": "typedef",
+    "value": {
+      "fill": {
+        "type": "color",
+        "required": false
+      },
+      "border": {
+        "color": {
+          "type": "color",
+          "required": false
+        },
+        "width": {
+          "type": "dimension",
+          "required": false
+        }
+      }
+    }
+  }
+}
+```
+
+</aside>
+
+It could then just save out that type definition into the token file it exports. Other tools could then make use of that same type definition and do useful things with tokens of that type.
+
+### Custom configuration
+
+Via configuration files or settings menus, tools could let users define custom behavior for types they have defined. For instance, in an export tool like Style Dictionary, users might be able to configure a custom transform or formatter for their composite types.
+Then, this example...
+
+<aside class="example" title="Custom configuration source">
+
+```json
+{
+  "My token": {
+    "type": "{my types.color pair}",
+    "value": {
+      "foreground": "#333333",
+      "background": "#ffffff"
+    }
+  }
+}
+```
+
+</aside>
+
+...could be configured to export, for example, a mixin instead of 2 variables:
+
+<aside class="example" title="Custom configuration Sass output">
+
+```css
+@mixin my-token {
+  color: #333333;
+  background-color: #ffffff;
+}
+```
+
+</aside>
+
+The downside is of course that teams need to set up and maintain these configurations themselves.
+
+### GUIs
+
+Since composite types are ultimately composed of the core design token types, a design tool could automatically display an appropriate UI for creating, editing, or previewing them.
+
+For instance, the color pair example above consists of 2 colors, one with the key “foreground” and the other with the key “background”. A design tool could therefore display two color picker widgets using those keys as the respective labels.
```

Comment:
should we use code wraps instead of quotes for the keys (foreground & background)?